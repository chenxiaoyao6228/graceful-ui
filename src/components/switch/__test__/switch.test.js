import { render, fireEvent } from "@testing-library/vue";
import Switch from "../switch.vue";
let spy = jest.fn();
let Component = {
  template: `<GSwitch v-model="value" @on-change="spy" data-testid="test"></GSwitch>`,
  components: { GSwitch: Switch },
  data() {
    return { value: false };
  },
  methods: {
    spy,
  },
};

describe("Switch", () => {
  it("should match snapshots", () => {
    const { container } = render(Component);
    expect(container).toMatchSnapshot();
  });

  it("toggle properly", async() => {
    const { getByTestId } = render(Component);
    let Switch = getByTestId("test");

    await fireEvent.click(Switch);
    expect(spy).toHaveBeenCalledTimes(1);
  });
});
