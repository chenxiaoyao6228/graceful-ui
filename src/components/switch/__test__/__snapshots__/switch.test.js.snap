// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch should match snapshots 1`] = `
<div>
  <label
    class="g-switch g-switch g-switch-default"
    data-testid="test"
  >
    <input
      type="hidden"
      value="false"
    />
     
    <span
      class="g-switch-inner"
    />
  </label>
</div>
`;
